### ISSUE TITLE
WebRTC requests are not filtered

### ISSUE THREAD

**Issue by nodiscc** [RequestPolicyContinued/requestpolicy#586]
https://diafygi.github.io/webrtc-ips/ leaks LAN and WAN IP addresses through WebRTC. By looking at the code at https://github.com/diafygi/webrtc-ips, I see the STUN server used for the demo is `stun.services.mozilla.com`. The request does not show up in the request log.

As the destination is on another domain, it should be blocked in _default deny_ mode.

**Comment by myrdd** [comment]
I can confirm this bug. From what I've seen so far it seems like `shouldLoad` is not called for WebRTC requests. Maybe they can be intervened differently?

I can see this is a problem with "policeman" as well. @futpib do you think we could work together on a solution?

**Comment by myrdd** [comment]
I've just found out that also NoScript does not display `stun.services.mozilla.com`. As soon as you allow `diafygi.github.io`, you allow the stun service as well.

**Comment by myrdd** [comment]
The easiest way would be to set `media.peerconnection.enabled` to `false` – see https://support.mozilla.org/en-US/questions/963501
That would disable WebRTC in general. If there is no other way, RP should encourage to disable WebRTC.

I've just discovered this Mozilla Bug: https://bugzilla.mozilla.org/show_bug.cgi?id=959893
I'm going to read that.

**Comment by futpib** [comment]
I looked for legit ways to intercept those connections and found none too. Right now it seems that the only way to block that selectively is to shim the WebRTC API on content `window` objects (like replace the `RTCPeerConnection` function with our own that first checks if it's allowed to work). Nice guys on irc suggested that [frame scripts](https://developer.mozilla.org/en-US/Firefox/Multiprocess_Firefox/The_message_manager#Loading_frame_scripts) might be the right way to do that, but I didn't tried it yet.

**Comment by myrdd** [comment]
Wow it's a really heated discussion right now at [Bug 959893](https://bugzilla.mozilla.org/show_bug.cgi?id=959893). It seems like it will be marked as wontfix, see 	[comment \#45](https://bugzilla.mozilla.org/show_bug.cgi?id=959893#c45).

What I've extracted is the following, written in [comment \#6](https://bugzilla.mozilla.org/show_bug.cgi?id=959893#c6):
> It should also be easy to build an extension to Firefox that adds a permission prompt to all PeerConnection requests (perhaps with the option to filter certain candidates, or limit all candidates to TURN).

I don't know what Randell meant by that, but what you @futpib have suggested seems to be one solution.

Btw does policeman currently wrap any of the browser's functions? RP does, I wondered whether it's necessary; I didn't investigate it enough yet.

> Nice guys on irc suggested that frame scripts might be the right way to do that, but I didn't tried it yet.

(some off-topic) I've just looked at your code the other day, looking for how you implement framescripts – you don't have any framescript yet, no? Maybe you've noticed that I've been working hard on getting RP restartless and to introduce framescripts. If you want to implement them it might help you to kow how I've implemented them.
